src/main/java/sirius/biz/jobs/params/Parameter.java

```diff
@@ -127,6 +129,21 @@ public boolean isVisible(Map<String, String> context) {
      */
     public abstract String getTemplateName();
 
+    /**
+     * Returns the name of the template used to render the parameter in the UI.
+     * <p>
+     * Similar to {@link #getTemplateName()}, but this method considers the visibility
+     * of the parameter and delivers an alternative template in case the parameter should be hidden.
+     *
+     * @return the name or path of the template used to render the parameter
+     */
+    public String computeTemplateName() {
+        if (this.visibility == Visibility.HIDDEN) {
```

Comment:
shouldn't this also take "ONLY_WITH_VALUE" into account?

Reply:
i.e. invoke isVisible

Reply:
Validating with isVisible()